integration_tests/tests/integration/common.rs

```diff
@@ -22,133 +22,92 @@ use std::{iter, time::Duration as StdDuration};
-    collector_max_elapsed_time: time::Duration,
-) -> (TaskParameters, TaskBuilder) {
-    let endpoint_fragments = EndpointFragments {
-        leader: AggregatorEndpointFragments::Remote {
-            url: leader_endpoint.clone(),
-        },
-        helper: AggregatorEndpointFragments::Remote {
-            url: helper_endpoint.clone(),
-        },
+    let (leader_endpoint, helper_endpoint, endpoint_fragments) = match test_context {
+        TestContext::VirtualNetwork => {
+            let endpoint_random_value = hex::encode(random::<[u8; 4]>());
+            (
+                Url::parse(&format!("http://leader-{endpoint_random_value}:8080/")).unwrap(),
+                Url::parse(&format!("http://helper-{endpoint_random_value}:8080/")).unwrap(),
+                EndpointFragments {
+                    leader: AggregatorEndpointFragments::VirtualNetwork {
+                        host: format!("leader-{endpoint_random_value}"),
+                        path: "/".to_string(),
+                    },
+                    helper: AggregatorEndpointFragments::VirtualNetwork {
+                        host: format!("helper-{endpoint_random_value}"),
+                        path: "/".to_string(),
+                    },
+                },
+            )
+        }
+        TestContext::Host => (
+            Url::parse("http://invalid/").unwrap(),
+            Url::parse("http://invalid/").unwrap(),
+            EndpointFragments {
+                leader: AggregatorEndpointFragments::Localhost {
+                    path: "/".to_string(),
+                },
+                helper: AggregatorEndpointFragments::Localhost {
+                    path: "/".to_string(),
+                },
+            },
+        ),
+        #[cfg(feature = "in-cluster")]
+        TestContext::Remote => (
+            task_builder.leader_aggregator_endpoint().clone(),
+            task_builder.helper_aggregator_endpoint().clone(),
+            EndpointFragments {
+                leader: AggregatorEndpointFragments::Remote {
+                    url: task_builder.leader_aggregator_endpoint().clone(),
+                },
+                helper: AggregatorEndpointFragments::Remote {
+                    url: task_builder.helper_aggregator_endpoint().clone(),
+                },
+            },
+        ),
     };
 
-    test_task_builder_inner(
-        vdaf,
-        query_type,
-        leader_endpoint,
-        helper_endpoint,
-        endpoint_fragments,
-        false,
-        collector_max_interval,
-        collector_max_elapsed_time,
-    )
-}
+    task_builder = task_builder
+        .with_leader_aggregator_endpoint(leader_endpoint)
+        .with_helper_aggregator_endpoint(helper_endpoint)
+        .with_min_batch_size(46)
+        // TODO(timg): is it OK to unconditionally set these? I think the remote case will ignore
+        // the values anyway
+        .with_dap_auth_aggregator_token()
+        .with_dap_auth_collector_token();
```

Comment:
I think you're right, although it's non-obvious. I'd just leave a comment to the effect of "we expect that the remote case ignores these tokens and substitutes its own".

Reply:
Oh, whoops, I meant to run `cloud-integration-test` against this branch, see what happens and update the comment accordingly. Thanks for flagging!

https://github.com/divviup/janus-ops/actions/runs/8149361321

Reply:
The test run failed, but that's because this DAP-09 Janus branch won't work with the DAP-07 staging environment. The test got far enough to fail at report decryption, which means that the authentication tokens were validated, and we were right about being able to unconditionally generate the tokens.